plugins/chat/assets/javascripts/discourse/components/chat-message-separator-date.js

```diff
@@ -0,0 +1,11 @@
+import Component from "@glimmer/component";
+import { action } from "@ember/object";
+
+export default class ChatMessageSeparatorDate extends Component {
+  @action
+  onDateClick() {
+    return this.args?.fetchMessagesByDate(
```

Comment:
`this.args` will always exist. Did you mean to check for existence of the function?

```suggestion
    return this.args.fetchMessagesByDate?.(
```

Reply:
yes indeed :thinking: